backend/server.js

```diff
@@ -23,6 +83,81 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+app.get('/thoughts', authenticateUser)
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find()
+  res.json({success: true, thoughts})
+})
+
+app.post('/thoughts', (req, res) => {
+  const { username, message } = req.body
+  
+  try {
+    const newThought = new Thought({
+      username, 
+      message
+    })
+    newThought.save()
+    res.status(201).json({success: true, id: newThought._id, username: newThought.username, createdAt: newThought.createdAt, message: newThought.message})
+  } catch (error) {
+    res.status(400).json({success: false, message: 'Could not post thought', error})
+  }
+  
+})
+
+app.post('/signup', async (req, res) => {
+  const { username, email, password } = req.body
+
+  try {  
+    const salt = bcrypt.genSaltSync()
+
+    const user = new User({
+      username, 
+      email, 
+      password: bcrypt.hashSync(password, salt)
+    }).save()
+    
+    res.status(201).json({success: true, id: user._id, username: user.username, email: user.email, accessToken: user.accessToken})
+    
+    
+    
+  } catch (error) {
+    res.status(400).json({success: false, message: 'Could not create user', error})
```

Comment:
There is no feedback for the user if it is successful or not, so it would be good to show for the user both if it is successful or not. For example if the user tries to create an account with an already existing username.
If the user successfully creates an account maybe he/she should be redirected to login for example?